catalyst_voices_packages/catalyst_cardano_serialization/lib/src/types.dart

```diff
@@ -3,6 +3,15 @@ import 'package:cbor/cbor.dart';
 import 'package:convert/convert.dart';
 import 'package:equatable/equatable.dart';
 
+/// An interface for classes that support CBOR serialization.
+abstract class CborEncodable extends Equatable {
```

Comment:
@ilap 
What do you think about separating concerns for this interface so that encoding (`toCbor`) is not tied up with data identity (`hashCode`, `equals`, `toString`)?

I would recommend creating a pure `CborEncodable` interface without extending from `Equatable`. There are few cases when a class is cbor encodable but it doesn't intend to extend from Equatable. For instance have a look at `BaseCertificate` which provides it's own `equals`, `hashCode` and `toString` but also has `toCbor` encoding. 

Consider to mark the CborEncodable as abstract interface to enforce using `implements` instead of `extends`.

Reply:
Hi @dtscalac,

Agree, you're right. I missed the `BaseCertificate` and some extensions' `toCbor()` functions.

Reply:
BTW @dtscalac @ilap 
what do you think to also apply `interface` class modifier ?
```Dart
abstract interface class CborEncodable
```

Reply:
Oooh, I was not aware this new class modifier which luckily has arrived to the Dart world.
It's definitely the better candidate if you guys choose the `implements` path and exposing some api to some external/other libraries.